osu.Game/Screens/OnlinePlay/DrawableRoomPlaylistItem.cs

```diff
@@ -87,25 +92,24 @@ protected override void LoadComplete()
 
             SelectedItem.BindValueChanged(selected => maskingContainer.BorderThickness = selected.NewValue == Model ? 5 : 0, true);
 
-            beatmap.BindValueChanged(_ => scheduleRefresh());
-            ruleset.BindValueChanged(_ => scheduleRefresh());
-
-            requiredMods.CollectionChanged += (_, __) => scheduleRefresh();
+            beatmap.BindValueChanged(_ => Scheduler.AddOnce(refresh));
+            ruleset.BindValueChanged(_ => Scheduler.AddOnce(refresh));
+            valid.BindValueChanged(_ => Scheduler.AddOnce(refresh));
+            requiredMods.CollectionChanged += (_, __) => Scheduler.AddOnce(refresh);
 
             refresh();
         }
 
-        private ScheduledDelegate scheduledRefresh;
         private PanelBackground panelBackground;
 
-        private void scheduleRefresh()
-        {
-            scheduledRefresh?.Cancel();
-            scheduledRefresh = Schedule(refresh);
-        }
-
         private void refresh()
         {
+            if (!valid.Value)
+            {
+                maskingContainer.BorderThickness = 5;
+                maskingContainer.BorderColour = colours.Red;
+            }
```

Comment:
If after an item being marked invalid another valid item is selected, the invalid red border disappears:

https://user-images.githubusercontent.com/20418176/142062897-a34856d9-a7bd-46b1-b823-a44cb3cffac9.mp4

Given that the invalid item is unclickable, that seems not intentional.

Reply:
Is this a possible scenario in gameplay?

Reply:
I've added logic to make this work more as expected, though I'm not sure it will ever be seen.

Reply:
As to whether this could be seen, I tested and could break it in game too. Unless I'm very confused, the initial room creation subscreen didn't/doesn't disable selection.

Reply:
The more you know... I've never actually selected an item there because... it does nothing 😅.